### ISSUE TITLE
Layer normalization for Nested Tensors does not return the correct tensor types

### ISSUE THREAD

**Issue by jananisriram** [pytorch/pytorch#133138]
### 🐛 Describe the bug

Layer normalization on `NestedTensor` returns the result of layer normalization, mean, and standard deviation. It should return 3 NTs if not reducing on the ragged dimension, and 3 dense tensors otherwise.

e.g.
```
import torch

size = 1024

lengths = torch.randint(10000, 16000, (size,), device="cpu")
offsets = torch.zeros((size + 1,), device="cpu", dtype=torch.int32)
torch.cumsum(lengths, dim=0, out=offsets[1:])

values = torch.randn(
    (offsets[-1].item(), size), device="cpu", requires_grad=True
)

nt = NestedTensor(values, offsets)

out = torch.nn.functional.layer_norm(nt, nt.shape[1:])
```

### Versions

fbcode H100 8/9/24
